crates/xflags-macros/tests/it/empty.rs

```diff
@@ -40,26 +40,26 @@ impl Empty {
         #![allow(non_snake_case, unused_mut)]
 
         let mut state_ = 0u8;
-        while let Some(arg_) = p_.pop_flag() {
+        if let Some(arg_) = p_.pop_flag() {
             match arg_ {
                 Ok(flag_) => match (state_, flag_.as_str()) {
                     (0, "--help" | "-h") => return Err(p_.help(Self::HELP_)),
-                    _ => return Err(p_.unexpected_flag(&flag_)),
```

Comment:
I'd avoid auto-chaining help message (unless fuchsia guidelines explicitly require that): my overall rule of thumb is to produce as little output as possible, as human attention is scarce. If they need a help message, the user can run `--help` themselves!

Reply:
I think not auto chaining is completely fine for flags and args actually, for missing commands it might still be a good idea to do it?

Reply:
>for missing commands it might still be a good idea to do it?

Hm, I think, as a user, I'd rather see

"unknown subcommand: `x`, existing subcommands: `u`, `r`, or `w`"

or

"subcommand requireq: `u`, `r`, or `w`"

or some such, rather than the entire help?

Reply:
For these, I think I'd have to either generate a bunch of these or generate them on the fly, neither of which feels great. Another issue is that it becomes quite a mess if you have 4 commands and 5 flags, you get a lot at a time. Maybe I can just do return an "unknown subcommand/flag `x`, use `help` for more information."? Would reduce the complexity of the whole thing quite a bit